build-plugin/src/main/kotlin/com/android/developers/androidify/plugin/AndroidLibraryPlugin.kt

```diff
@@ -0,0 +1,43 @@
+package com.android.developers.androidify.plugin
+
+import com.android.build.api.dsl.LibraryExtension
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.assign
+import org.gradle.kotlin.dsl.configure
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.dsl.KotlinAndroidProjectExtension
+
+class AndroidLibraryPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        with(project) {
+            apply(plugin = "com.android.library")
+            apply(plugin = "org.jetbrains.kotlin.android")
+
+            extensions.configure<LibraryExtension> {
+                compileSdk = getVersionByName("compileSdk").toInt()
```

Comment:
this is not idea. I'd use a settings plugin for these values. Putting them in the catalog is not best practice.